cparser/Elab.ml

```diff
@@ -1386,7 +1417,7 @@ let elab_expr vararg loc env a =
         | TUnion(id, attrs) ->
             (wrap Env.find_union_member loc env (id, fieldname), attrs)
         | _ ->
-            error "left-hand side of '.' is not a struct or union" in
+            error "member reference base type %a is not a structure or union" (print_typ env) b1.etyp in
```

Comment:
The new message (clang's) is quite obscure.  GCC is better:
```
gee.c:3:24: error: request for member ‘a’ in something not a structure or union
```
The original message wasn't too bad either.